tfmigrate/migrator.go

```diff
@@ -30,11 +31,21 @@ func setupWorkDir(ctx context.Context, tf tfexec.TerraformCLI, workspace string,
 	}
 	log.Printf("[INFO] [migrator@%s] terraform version: %s\n", tf.Dir(), version)
 
+	supportsStateReplaceProvider, constraints, err := tf.SupportsStateReplaceProvider(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
 	// init folder
 	log.Printf("[INFO] [migrator@%s] initialize work dir\n", tf.Dir())
 	err = tf.Init(ctx, "-input=false", "-no-color")
 	if err != nil {
-		return nil, nil, err
+		acceptable := "Error: Invalid legacy provider address"
+		if supportsStateReplaceProvider && strings.Contains(err.Error(), acceptable) {
+			log.Printf("[INFO] [migrator@%s] ignoring error '%s' initilizing work dir; the error is expected when using Terraform %s with a legacy Terraform state\n", tf.Dir(), acceptable, constraints)
```

Comment:
☝️ Could/should this logic be improved? For example, perhaps the `Error: Invalid legacy provider address` should only be permitted during `state replace-provider` actions, but not other actions? Doing so would require passing the action(s) to `setupWorkDir`, though. Do you have thoughts, @minamijoyo ?

Reply:
I think it is safer to narrow down the special error-handling conditions.
Rather than passing actions directly to the setupWorkDir function, why not pass it as a bool argument, such as ignoreLegacyState?

Reply:
@minamijoyo Yep; we can pass a `bool` flag if you prefer. I opted not to, as that arguably redundantly distributes the need to correctly set this `bool` value across all invocations of the function (rather than consolidated the logic to a single place, within the function). However, perhaps you're right that doing so is best in this case. Stay tuned :)

Reply:
@minamijoyo I believe I've implemented your suggestion in https://github.com/minamijoyo/tfmigrate/pull/145/commits/67ed6758bc45d3c881e75c5c56a3f5d497913cc5. Does that seem acceptable?